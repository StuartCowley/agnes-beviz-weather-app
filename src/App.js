import "./App.css";
import React from "react";

// <App />
function App() {
  return (
    <div className="App">
      <h1>Weather App</h1>
      <h2>{2 + 2}</h2>
    </div>
  );
}

export default App;

// react component example
// const App = () => <h1>Hello World</h1>;

// The above utilises an implicit return, which is a feature of arrow-functions.
// It is the same as this:
// const App = () => {
//  return <h1>Hello World</h1>;
// }
// or this:
// function App() {
//   return <h1>Hello World</h1>;
// }

// export default App;
