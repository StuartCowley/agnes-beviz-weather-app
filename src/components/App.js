import "../styles/App.css";
import React from "react";
import PropTypes from "prop-types";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";

// <App />
function App({ location, forecasts }) {
  const { city, country } = location;
  return (
    <div className="App">
      <LocationDetails city={city} country={country} />
      <ForecastSummaries forecasts={forecasts} />
    </div>
  );
}

App.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
  location: PropTypes.shape({
    city: PropTypes.string,
    country: PropTypes.string,
  }).isRequired,
};

export default App;

// react component example
// const App = () => <h1>Hello World</h1>;

// The above utilises an implicit return, which is a feature of arrow-functions.
// It is the same as this:
// const App = () => {
//  return <h1>Hello World</h1>;
// }
// or this:
// function App() {
//   return <h1>Hello World</h1>;
// }

// export default App;
