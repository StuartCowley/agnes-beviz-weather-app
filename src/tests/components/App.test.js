import { getByTestId, render, screen } from "@testing-library/react";
import React from "react";
import App from "../../components/App";
// import forecast from "../../data/forecast.json";

describe("App", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();

    /* expect(getByTestId("h1-LocationDetails")).toBeInstanceOf(
      HTMLHeadingElement
    ); */
  });
});

// render(<App />);
// const h1Element = screen.getByText(/Manchester, UK/i);

// expect(h1Element).toBeInTheDocument();
