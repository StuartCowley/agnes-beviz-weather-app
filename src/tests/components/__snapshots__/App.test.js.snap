// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      data-testid="h1-LocationDetails"
    >
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
